llvm/test/CodeGen/WebAssembly/offset.ll

```diff
@@ -60,6 +60,29 @@ define i32 @load_i32_with_folded_gep_offset_nuw(ptr %p) {
   ret i32 %t
 }
 
+@global_data = external global i32
+
+define i32 @load_i32_global_with_folded_gep_offset_nonconst_nuw(i32 %idx) {
+; CHECK-LABEL: load_i32_global_with_folded_gep_offset_nonconst_nuw:
+; CHECK:    i32.const $push0=, 2
+; CHECK-NEXT:    i32.shl $push1=, $0, $pop0
+; CHECK-NOT:     i32.add
+; CHECK-NEXT:    i32.load $push2=, global_data($pop1)
+  %s = getelementptr nuw i32, ptr @global_data, i32 %idx
+  %t = load i32, ptr %s
+  ret i32 %t
+}
+
+define i32 @load_i32_global_with_folded_gep_offset_const_nuw() {
+; CHECK-LABEL: load_i32_global_with_folded_gep_offset_const_nuw:
+; CHECK:    i32.const $push0=, 8
+; CHECK-NEXT:    i32.load $push1=, global_data($pop0)
+entry:
+  %t = load i32, ptr getelementptr nuw (i32, ptr @global_data, i32 2)
+  ret i32 %t
+}
```

Comment:
I don't think we really need to test constant expressions actually since nowhere else in this file tests them, i think you can just delete this test